### PR TITLE
fix(menu): updated bg color for legacy theme

### PR DESCRIPTION
**Pull request by TarunAdobe**
## Description

This PR updates the background-color-default for menu-item in legacy theme.

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

 - [ ] No expected changes to `metadata.json`

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [ ] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [ ] VRTs have been run and looked at.
- [ ] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.

## Screenshots

Previously:

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [ ] I have updated relevant storybook stories and templates.
- [ ] I have tested these changes in Windows High Contrast mode.
- [ ] If my change impacts **other components**, I have tested to make sure they don't break.
- [ ] If my change impacts **documentation**, I have updated the documentation accordingly.
- [ ] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

File: components/menu/themes/spectrum.css [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 @container style(--system: legacy) {
 	.spectrum-Menu {
-		--spectrum-menu-item-background-color-default: var(--spectrum-gray-50);
+		--spectrum-menu-item-background-color-default: var(--spectrum-gray-100);
```
Comment (marissahuysentruyt) [inline components/menu/themes/spectrum.css:20]: With this change, what should I expect to see with the hovers? I don't visually see a difference anymore, but maybe I'm not supposed to yet.

Comment (marissahuysentruyt) [inline components/menu/themes/spectrum.css:20]: I don't think I see a change visually now when a menu item is hovered. Should I be expecting one?

Comment (castastrophe) [reply components/menu/themes/spectrum.css:20]: The styles for `--spectrum-menu-item-background-color-default` on the current S1 branch are actually [transparent](--spectrum-menu-item-background-color-default). I wonder if we should be correcting this to be:
 
```suggestion
		--spectrum-menu-item-background-color-default: var(--spectrum-transparent-black-100);
```

**Review by castastrophe — CHANGES_REQUESTED** [review]

**Comment by Rajdeepc** [comment]
Reopened in a clean #3240

**Pull request closed without merging**